ambry-clustermap/src/main/java/com/github/ambry/clustermap/Partition.java

```diff
@@ -110,6 +112,21 @@ public Partition(long id, String partitionClass, PartitionState partitionState,
     return Collections.unmodifiableList(result);
   }
 
+  @Override
+  public Map<ReplicaState, List<ReplicaId>> getReplicaIdsByStates(Set<ReplicaState> states, String dcName) {
+    Map<ReplicaState, List<ReplicaId>> replicaByState = new HashMap<>();
+    for (ReplicaState state : states) {
+      List<ReplicaId> replicaIds = new ArrayList<>();
+      if (state == ReplicaState.STANDBY) {
```

Comment:
Why are all replicas added when the state STANDBY is requested?

Reply:
Oh, I should add a comment there. This is because `Partition` is used by static clustermap where all replicas are assumed to be `STANDBY` and we leverage native failure detector to check their status (and backoff if needed).